File: core/src/com/symbol/input/KeyInput.java
package com.symbol.input;

import com.badlogic.gdx.Input;
import com.badlogic.gdx.InputProcessor;

public class KeyInput implements InputProcessor {

    private KeyInputHandler handler;

    public KeyInput(KeyInputHandler handler) {
        this.handler = handler;
    }

    @Override
    public boolean keyDown(int keycode) {
        switch (keycode) {
            case Input.Keys.RIGHT: handler.move(true); break;
            case Input.Keys.LEFT: handler.move(false); break;
            case Input.Keys.Z: handler.jump(); break;
            case Input.Keys.X: handler.shoot(); break;
        }
        return true;
    }

    @Override
    public boolean keyUp(int keycode) {
        switch (keycode) {
            case Input.Keys.RIGHT: handler.stop(true); break;
            case Input.Keys.LEFT: handler.stop(false); break;
        }
        return true;
    }

    @Override
    public boolean keyTyped(char character) { return false; }

    @Override
    public boolean touchDown(int screenX, int screenY, int pointer, int button) { return false; }

    @Override
    public boolean touchUp(int screenX, int screenY, int pointer, int button) { return false; }

    @Override
    public boolean touchDragged(int screenX, int screenY, int pointer) { return false; }

    @Override
    public boolean mouseMoved(int screenX, int screenY) { return false; }

    @Override
    public boolean scrolled(int amount) { return false; }

}

File: core/src/com/symbol/game/input/KeyInput.kt
package com.symbol.game.input

import com.badlogic.gdx.Input
import com.badlogic.gdx.InputProcessor

class KeyInput(private val handler: KeyInputHandler) : InputProcessor {

    override fun keyDown(keycode: Int): Boolean {
        when (keycode) {
            Input.Keys.RIGHT -> handler.move(true)
            Input.Keys.LEFT -> handler.move(false)
            Input.Keys.Z -> handler.jump()
            Input.Keys.X -> handler.shoot()
            Input.Keys.C -> handler.release()
        }
        return true
    }

    override fun keyUp(keycode: Int): Boolean {
        when (keycode) {
            Input.Keys.RIGHT -> handler.stop(true)
            Input.Keys.LEFT -> handler.stop(false)
        }
        return true
    }

    override fun keyTyped(character: Char): Boolean = true

    override fun touchUp(screenX: Int, screenY: Int, pointer: Int, button: Int): Boolean = false

    override fun mouseMoved(screenX: Int, screenY: Int): Boolean = false

    override fun scrolled(amount: Int): Boolean = false

    override fun touchDragged(screenX: Int, screenY: Int, pointer: Int): Boolean = false

    override fun touchDown(screenX: Int, screenY: Int, pointer: Int, button: Int): Boolean = false

}